extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchMergeAggregatorFactory.java

```diff
@@ -142,6 +151,22 @@ public VectorAggregator factorizeVector(VectorColumnSelectorFactory selectorFact
     );
   }
 
+  private void validateInputs(@Nullable ColumnCapabilities capabilities)
+  {
+    if (capabilities != null) {
+      final ColumnType type = capabilities.toColumnType();
+      if (!ColumnType.UNKNOWN_COMPLEX.equals(type) && !TYPE.equals(type) &&
+          !(ValueType.COMPLEX.equals(type.getType()) &&
+            (Objects.equals(type.getComplexTypeName(), "HLLSketch") ||
+             Objects.equals(type.getComplexTypeName(), "HLLSketchBuild")))) {
+        throw new UOE("Using aggregation type %s is not supported for %s column. "
+                      + "Use a different aggregator type and run the query again.",
+                      getIntermediateType().getComplexTypeName(),
```

Comment:
From the error message, this is suppposed to be an "aggregation type". This is the complex column type name, not the aggregation type.

Reply:
Sample error message: `Using aggregator [HLLSketchMerge] is not supported for complex columns with type [COMPLEX<json>].`

In the native query, values like HLLSketchMerge appear under `aggregations -> type` nesting, so it should be aggregation type? I'll rename it to aggregator as suggested though.
```json
      "aggregations": [
        {
          "type": "HLLSketchMerge",
          "name": "a0",
          "fieldName": "shipTo",
          "lgK": 12,
          "tgtHllType": "HLL_4",
          "round": true
        }
```
Thoughts?